src/base/registrations/registrar.ts

```diff
@@ -87,14 +91,19 @@ export async function registerName(name: string, owner: string, config: Config)
       await commitAndRegister(name, owner, config);
     }
   } catch (e) {
-    throw { type: Failure.TransactionFailed, message: e.message, txHash: e.txHash };
+    throw { type: e.type || Failure.TransactionFailed, message: e.message, txHash: e.txHash };
   }
 }
 
 async function commitAndRegister(name: string, owner: string, config: Config) {
   let salt = ethers.utils.randomBytes(32);
   let minAge = (await registrar(config).minCommitmentAge()).toNumber();
   let fee = await registrationFee(config);
+  // Avoids gas spent by the owner, trying to commit to a name and not having
+  // enough RAD balance
+  if ((await radToken(config).balanceOf(owner)).toNumber() < fee) {
```

Comment:
This can overflow if the balance is not representable as a javascript number. It's best to use the `.lt()` method on `BigNumber` to make the comparison.